File: ApiTester-DotNet/ApiTester-DotNet/ApiTester.cs
using System;
using System.Net.Http;
using System.Reflection.Emit;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using static System.Net.Mime.MediaTypeNames;

public class ApiTester : Form
{
    TextBox urlBox = new TextBox();
    ComboBox methodBox = new ComboBox();
    TextBox paramBox = new TextBox();
    TextBox resultBox = new TextBox();
    Button sendButton = new Button();

    public ApiTester()
    {
        this.Text = "API Test Tool - .NET Version";
        this.Size = new System.Drawing.Size(600, 500);

        System.Windows.Forms.Label urlLabel = new System.Windows.Forms.Label() { Text = "API URL:", Left = 10, Top = 10, Width = 80 };
        urlBox.Left = 100; urlBox.Top = 10; urlBox.Width = 480;

        System.Windows.Forms.Label methodLabel = new System.Windows.Forms.Label() { Text = "Method:", Left = 10, Top = 45, Width = 80 };
        methodBox.Left = 100; methodBox.Top = 45; methodBox.Width = 80;
        methodBox.Items.AddRange(new string[] { "GET", "POST" });
        methodBox.SelectedIndex = 0;

        System.Windows.Forms.Label paramLabel = new System.Windows.Forms.Label() { Text = "Request Params (JSON or Form):", Left = 10, Top = 80, Width = 220 };
        paramBox.Left = 10; paramBox.Top = 110; paramBox.Width = 570; paramBox.Height = 80; paramBox.Multiline = true;

        sendButton.Text = "Send Request"; sendButton.Left = 10; sendButton.Top = 200; sendButton.Width = 570;
        sendButton.Click += async (s, e) => await SendRequest();

        System.Windows.Forms.Label resultLabel = new System.Windows.Forms.Label() { Text = "Response:", Left = 10, Top = 240, Width = 100 };
        resultBox.Left = 10; resultBox.Top = 270; resultBox.Width = 570; resultBox.Height = 180; resultBox.Multiline = true; resultBox.ScrollBars = ScrollBars.Vertical;

        this.Controls.Add(urlLabel);
        this.Controls.Add(urlBox);
        this.Controls.Add(methodLabel);
        this.Controls.Add(methodBox);
        this.Controls.Add(paramLabel);
        this.Controls.Add(paramBox);
        this.Controls.Add(sendButton);
        this.Controls.Add(resultLabel);
        this.Controls.Add(resultBox);
    }

    private async Task SendRequest()
    {
        string url = urlBox.Text.Trim();
        string method = methodBox.SelectedItem.ToString();
        string param = paramBox.Text.Trim();

        resultBox.Text = "Requesting...";
        try
        {
            using (HttpClient client = new HttpClient())
            {
                HttpResponseMessage response;
                if (method == "GET")
                {
                    if (!string.IsNullOrEmpty(param))
                    {
                        url += url.Contains("?") ? "&" + param : "?" + param;
                    }
                    response = await client.GetAsync(url);
                }
                else // POST
                {
                    StringContent content = new StringContent(param, Encoding.UTF8, "application/json");
                    response = await client.PostAsync(url, content);
                }
                string result = await response.Content.ReadAsStringAsync();
                resultBox.Text = FormatJson(result);
            }
        }
        catch (Exception ex)
        {
            resultBox.Text = "Request error: " + ex.Message;
        }
    }

    // Simple JSON formatter
    private string FormatJson(string json)
    {
        try
        {
            int indent = 0;
            StringBuilder sb = new StringBuilder();
            foreach (char c in json)
            {
                if (c == '{' || c == '[')
                {
                    sb.Append(c).Append('\n');
                    indent++;
                    sb.Append(new string(' ', indent * 2));
                }
                else if (c == '}' || c == ']')
                {
                    sb.Append('\n');
                    indent--;
                    sb.Append(new string(' ', indent * 2)).Append(c);
                }
                else if (c == ',')
                {
                    sb.Append(c).Append('\n').Append(new string(' ', indent * 2));
                }
                else
                {
                    sb.Append(c);
                }
            }
            return sb.ToString();
        }
        catch
        {
            return json;
        }
    }
}

File: ApiTester-Java/src/ApiTester.java
import javax.swing.*;
import java.awt.event.ActionEvent;
import java.io.*;
import java.net.*;
import java.nio.charset.StandardCharsets;

public class ApiTester extends JFrame {
    private JTextField urlField = new JTextField();
    private JComboBox<String> methodBox = new JComboBox<>(new String[]{"GET", "POST"});
    private JTextArea paramArea = new JTextArea();
    private JTextArea resultArea = new JTextArea();
    private JButton sendButton = new JButton("Send Request");

    public ApiTester() {
        setTitle("API Testing Tool - Java Version");
        setSize(600, 500);
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        setLayout(null);

        JLabel urlLabel = new JLabel("API URL:");
        urlLabel.setBounds(10, 10, 80, 25);
        urlField.setBounds(100, 10, 480, 25);

        JLabel methodLabel = new JLabel("Method:");
        methodLabel.setBounds(10, 45, 80, 25);
        methodBox.setBounds(100, 45, 80, 25);

        JLabel paramLabel = new JLabel("Request Parameters (JSON or Form):");
        paramLabel.setBounds(10, 80, 250, 25);
        JScrollPane paramPane = new JScrollPane(paramArea);
        paramPane.setBounds(10, 110, 570, 80);

        sendButton.setBounds(10, 200, 570, 30);

        JLabel resultLabel = new JLabel("Response:");
        resultLabel.setBounds(10, 240, 100, 25);
        JScrollPane resultPane = new JScrollPane(resultArea);
        resultPane.setBounds(10, 270, 570, 180);

        add(urlLabel); add(urlField);
        add(methodLabel); add(methodBox);
        add(paramLabel); add(paramPane);
        add(sendButton);
        add(resultLabel); add(resultPane);

        sendButton.addActionListener(this::sendRequest);
    }

    private void sendRequest(ActionEvent e) {
        String url = urlField.getText().trim();
        String method = (String) methodBox.getSelectedItem();
        String params = paramArea.getText().trim();

        resultArea.setText("Requesting...");
        new Thread(() -> {
            try {
                String response = sendHttpRequest(url, method, params);
                resultArea.setText(formatJson(response));
            } catch (Exception ex) {
                resultArea.setText("Request failed: " + ex.getMessage());
            }
        }).start();
    }

    private String sendHttpRequest(String urlStr, String method, String params) throws Exception {
        if ("GET".equals(method) && !params.isEmpty()) {
            urlStr += urlStr.contains("?") ? "&" + params : "?" + params;
        }
        URL url = new URL(urlStr);
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod(method);
        conn.setConnectTimeout(5000);
        conn.setReadTimeout(5000);

        if ("POST".equals(method)) {
            conn.setDoOutput(true);
            try (OutputStream os = conn.getOutputStream()) {
                os.write(params.getBytes(StandardCharsets.UTF_8));
            }
        }
        int code = conn.getResponseCode();
        InputStream is = (code < 400) ? conn.getInputStream() : conn.getErrorStream();
        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
        StringBuilder sb = new StringBuilder();
        String line;
        while((line = reader.readLine()) != null) sb.append(line).append('\n');
        return sb.toString();
    }

    // Simple JSON formatter
    private String formatJson(String json) {
        try {
            int indent = 0;
            StringBuilder sb = new StringBuilder();
            for (char c : json.toCharArray()) {
                if (c == '{' || c == '[') {
                    sb.append(c).append('\n');
                    indent++;
                    sb.append("  ".repeat(indent));
                } else if (c == '}' || c == ']') {
                    sb.append('\n');
                    indent--;
                    sb.append("  ".repeat(indent)).append(c);
                } else if (c == ',') {
                    sb.append(c).append('\n').append("  ".repeat(indent));
                } else {
                    sb.append(c);
                }
            }
            return sb.toString();
        } catch (Exception e) {
            return json;
        }
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> new ApiTester().setVisible(true));
    }
}